homeassistant/components/knx/expose.py

```diff
@@ -11,29 +13,39 @@
     STATE_UNAVAILABLE,
     STATE_UNKNOWN,
 )
-from homeassistant.core import HomeAssistant, callback
+from homeassistant.core import callback
 from homeassistant.helpers.event import async_track_state_change_event
-from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.typing import (
+    ConfigType,
+    EventType,
+    HomeAssistantType,
+    StateType,
+)
 
 from .const import KNX_ADDRESS
 from .schema import ExposeSchema
 
 
 @callback
 def create_knx_exposure(
-    hass: HomeAssistant, xknx: XKNX, config: ConfigType
+    hass: HomeAssistantType, xknx: XKNX, config: ConfigType
 ) -> KNXExposeSensor | KNXExposeTime:
     """Create exposures from config."""
     address = config[KNX_ADDRESS]
+    expose_type = config[ExposeSchema.CONF_KNX_EXPOSE_TYPE]
     attribute = config.get(ExposeSchema.CONF_KNX_EXPOSE_ATTRIBUTE)
     entity_id = config.get(CONF_ENTITY_ID)
-    expose_type = config.get(ExposeSchema.CONF_KNX_EXPOSE_TYPE)
     default = config.get(ExposeSchema.CONF_KNX_EXPOSE_DEFAULT)
 
     exposure: KNXExposeSensor | KNXExposeTime
-    if expose_type.lower() in ["time", "date", "datetime"]:
+    if isinstance(expose_type, str) and expose_type.lower() in [
+        "time",
+        "date",
+        "datetime",
+    ]:
         exposure = KNXExposeTime(xknx, expose_type, address)
     else:
+        assert entity_id is not None
```

Comment:
Asserts don't work in an actual production environment. So in this case, either the above:

```
entity_id = config.get(CONF_ENTITY_ID)
```

Is incorrect, or in this place, an actual guard must be placed.

Reply:
CONF_ENTITY_ID is required now for non-time exposures. This would be cleaner if we separate time exposures form sensor/entity exposures completely in configuration, but this is for another PR as it would break configs.